### PR TITLE
[capstone] Update Capstone to v4.0.1-49-ge3d5a594

### LINKED ISSUE #5276: Update Capstone library
Current release is 4: https://github.com/aquynh/capstone/releases
vcpkg supports v3

### PR DESCRIPTION
**Pull request by ceeac**
I opted to base the package on commit e3d5a594 (latest v4 branch as of today) rather than v4.0.1 because v4.0.1 does not build successfully with x86_reduce enabled.

This PR also adds support for selectively enabling architecture support and/or diet mode.

Fixes #5276.

### CODE REVIEW

**Comment by Rastaban** [comment]
It looks like there are new build breaks on the UWP triplets on windows.  Would you be able to look into those?

File: ports/capstone/portfile.cmake [hunk 1]
```diff
@@ -1,36 +1,71 @@
 include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REF 3.0.5-rc3
+    REF e3d5a594
```
Comment (Rastaban) [inline ports/capstone/portfile.cmake]: Could you use the full sha here?

Comment (ceeac) [reply ports/capstone/portfile.cmake]: Done.

**Comment by ceeac** [comment]
The failing UWP build is caused by a bug in upstream Capstone; aquynh/capstone#1421 should fix this. I'll wait for the upstream PR to be merged and then update straight to the newest v4 branch once the PR is merged.

**Comment by Rastaban** [comment]
Sounds good, I will watch for your update.

**Pull request merged**